src/components/NcDialog/NcDialog.vue

```diff
@@ -0,0 +1,352 @@
+				<nav v-if="hasNavigation" :class="['dialog__navigation', ...navigationClasses]">
+					<slot name="navigation" :is-collapsed="isNavigationCollapsed" />
+				</nav>
+				<!-- Main dialog content -->
+				<div class="dialog__content">
+					<!-- If the navigation is shown on the side the header should be directly aligned with the content -->
+					<h2 v-if="!isNavigationCollapsed" class="dialog__name">
+						{{ props.name }}
+					</h2>
+					<slot>
+						<p>{{ props.message }}</p>
+					</slot>
+				</div>
+			</div>
+			<!-- The dialog actions aka the buttons -->
+			<div class="dialog__actions">
+				<slot name="actions">
+					<NcDialogButton v-for="(button, idx) in props.buttons"
+						:key="`${button.label}${idx}`"
+						v-bind="button"
+						@click="handleButtonClose" />
+				</slot>
+			</div>
+		</Fragment>
+	</NcModal>
+</template>
+
+<script setup lang="ts">
+// eslint-disable-next-line n/no-missing-import
+import type { IDialogButton } from './types'
+
+import { useElementSize } from '@vueuse/core'
+import { computed, ref, useSlots } from 'vue'
+import { Fragment } from 'vue-frag' // can be dropped with vue3
+
+import NcModal from '../NcModal/index.js'
+import NcDialogButton from './NcDialogButton.vue'
+
+const props = withDefaults(defineProps<{
+	/** Name of the dialog (the heading) */
+	name: string
+	/** Text of the dialog */
+	message?: string
+	/**
+	 * The element here to mount the dialog
+	 * @default 'body'
+	 */
+	container?: string
+	/**
+	 * Size of the underlying NcModal
+	 * @default 'normal'
+	 */
+	size?: 'small' | 'normal' | 'large' | 'full'
+	/**
+	 * Array of buttons to display
+	 * @see NcDialogButton for supported properties of a button
+	 * @default []
+	 */
+	buttons?: readonly IDialogButton[]
+	/**
+	 * Whether the dialog should be shown
+	 * @default true
+	 */
+	open?: boolean
+	/**
+	 * Declare if hiding the modal should be animated
+	 * @default false
+	 */
+	outTransition?: boolean
+	/**
+	 * Set to false to no show a close button on the dialog
+	 * @default true
+	 */
+	canClose?: boolean
+	/**
+	 * Optionally pass additionaly classes which will be set on the navigation for custom styling
+	 * @default []
+	 * @example
+	 * ```html
+	 * <DialogBase :navigation-classes="['mydialog-navigation']"><!-- --></DialogBase>
+	 * <!-- ... -->
+	 * <style lang="scss">
+	 * :deep(.mydialog-navigation) {
+	 *     flex-direction: row-reverse;
+	 * }
+	 * </style>
+	 * ```
+	 */
+	navigationClasses?: string[]
+}>(), {
+	size: 'normal',
+	container: 'body',
+	message: '',
+	buttons: () => [],
+	canClose: true,
+	open: true,
+	outTransition: false,
+	navigationClasses: () => [],
+})
+
+const emit = defineEmits<{
+	/** Emitted when the dialog is about to close but the out transition did not finished yet */
+	(e: 'closing'): void
+	/** Emitted when the open state changed */
+	(e: 'update:open', v: boolean): void
+}>()
+
+const slots = useSlots()
```

Comment:
I don't remember if it is supported in Vue 2.7, but in `SFC Setup` there is a `defineSlots` macro to define slots with TS.